Tests/MessagingInApp/Core/IntegrationTest.swift

```diff
@@ -38,12 +38,97 @@ open class IntegrationTest: UnitTest {
             completionHandler(.success((body, response)))
         }
     }
+
+    // Given a list of messages, this function sets up a HTTP response for the fetch user queue request.
+    func setupMockFetchResponse(messagesFetched messages: [Message]) {
+        if messages.isEmpty {
+            setupHttpResponse(code: 200, body: "[]".data)
+            return
+        }
+
+        // Construct a JSON string that will become the HTTP response body.
+        // The format of the JSON is the same as what /api/v1/users endpoint returns from Gist backend.
+        //
+        // We are opting to construct JSON strings manually with a dictionary because the data types used for parsing HTTP response bodies, `UserQueueResponse`, is not easily able to inherit the `Codable` protocol because of `[String: Any]`.
```

Comment:
Constructing JSON strings using a dictionary isn't ideal. I prefer the method used in Track v1 tests, where we construct Swift objects and generate JSON strings from them. However, the in-app SDK data types can't easily be made `Codable`.

Since the in-app SDK only performs this 1 HTTP request, I don't consider this a big piece of tech debt in the codebase. I can write a ticket to improve this in the future if others feel this is important.